### PR TITLE
Acceptable request

### PR DESCRIPTION
**Pull request by shjortConcordium**
## Purpose

Add bindings for acceptable request / statement from concordium-base wallet_library.

## Changes

_Describe the changes that were needed.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: concordium-sdk/src/main/java/com/concordium/sdk/crypto/CryptoJniNative.java [hunk 1]
```diff
@@ -316,4 +318,14 @@ public static String getVerifiableCredentialBackupEncryptionKey(String seedAsHex
      * {@link Web3IdProof} as JSON.
      */
     public static native String createWeb3IdProof(String input);
+
+    /**
+     * Check that a request is acceptable
+     * @param input {@link Request} serialized as JSON.
+     * @return JSON representing {@link StringResult}. If successful the field 'result' is empty,
```
Comment (orhoj) [inline concordium-sdk/src/main/java/com/concordium/sdk/crypto/CryptoJniNative.java:325]: You are kind of abusing `StringResult` if this is what you are doing. You could make a type specifically for this.

File: concordium-sdk/src/main/java/com/concordium/sdk/crypto/CryptoJniNative.java [hunk 2]
```diff
@@ -316,4 +318,14 @@ public static String getVerifiableCredentialBackupEncryptionKey(String seedAsHex
...
+
+    /**
+     * Check that a request is acceptable
+     * @param input {@link Request} serialized as JSON.
+     * @return JSON representing {@link StringResult}. If successful the field 'result' is empty, 
+     * but if not acceptable the reason will be contained in the error.
+     */
+    public static native String isAcceptableRequest(String input);
+
+    public static native String isAcceptableAtomicStatement(String input, String rangeTags, String setTags, AcceptableRequest.RawAttributeCheck check);
```
Comment (orhoj) [inline concordium-sdk/src/main/java/com/concordium/sdk/crypto/CryptoJniNative.java:330]: Missing javadoc.

File: crypto-jni/src/lib.rs [hunk 3]
```diff
@@ -1314,3 +1327,140 @@ pub extern "system" fn Java_com_concordium_sdk_crypto_CryptoJniNative_createWeb3
...
+
+    match request.acceptable_request(&wallet_library::default_wallet_config::default_wallet_config()) {
+        Ok(r) => r,
+        Err(err) => return StringResult::from(err).to_jstring(&env),
+    };
+
+    CryptoJniResult::Ok(()).to_jstring(&env)
+}
+
+/// The JNI wrapper for checking that a request is acceptable.
```
Comment (orhoj) [inline crypto-jni/src/lib.rs:1369]: The docs here are the same as for the other method.

File: crypto-jni/src/lib.rs [hunk 4]
```diff
@@ -1314,3 +1327,140 @@ pub extern "system" fn Java_com_concordium_sdk_crypto_CryptoJniNative_createWeb3
...
+    range_tags_input: JString,
+    set_tags_input: JString,
+    attribute_check: JObject<'a>,
+) -> jstring {
+    let input_string = match get_string(env, request_input) {
+        Ok(s) => s,
+        Err(err) => return StringResult::Err(err).to_jstring(&env),
+    };
+
+    let request: AtomicStatement<constants::ArCurve, String, Web3IdAttribute> =
```
Comment (orhoj) [inline crypto-jni/src/lib.rs:1388]: Not a request, rename the variable.

File: crypto-jni/src/lib.rs [hunk 5]
```diff
@@ -1314,3 +1327,140 @@ pub extern "system" fn Java_com_concordium_sdk_crypto_CryptoJniNative_createWeb3
...
+/// The JNI wrapper for checking that a request is acceptable.
+/// * `input` - the JSON string of [`wallet_library::proofs::RequestStatement`]
+#[no_mangle]
+#[allow(non_snake_case)]
+pub extern "system" fn Java_com_concordium_sdk_crypto_CryptoJniNative_isAcceptableAtomicStatement<
+    'a,
+>(
+    env: JNIEnv<'a>,
+    _: JClass,
+    request_input: JString,
```
Comment (orhoj) [inline crypto-jni/src/lib.rs:1378]: statement_input

**Review by orhoj — CHANGES_REQUESTED** [review]

File: concordium-sdk/src/main/java/com/concordium/sdk/crypto/wallet/web3Id/AcceptableRequest.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+import com.fasterxml.jackson.databind.JsonMappingException;
+
+public class AcceptableRequest {
+    // Static block to load native library.
+    static {
+        NativeResolver.loadLib();
+    }
+
+    /**
+     * Check that a request is acceptable
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply concordium-sdk/src/main/java/com/concordium/sdk/crypto/wallet/web3Id/AcceptableRequest.java]: I've tried to expand on it.

File: concordium-sdk/src/main/java/com/concordium/sdk/crypto/wallet/web3Id/AcceptableRequest.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+     *
+     * @param input the request that should be checked
+     * @return a Presentation serialized as JSON
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply concordium-sdk/src/main/java/com/concordium/sdk/crypto/wallet/web3Id/AcceptableRequest.java]: I've changed it to use a specific exception instead. :thinking:

**Review by orhoj — APPROVED** [review]

**Pull request merged**